### PR TITLE
Adding note on how to use datastore cluster for volume provisioning

### PR DESCRIPTION
**Pull request by divyenpatel**
**What this PR does / why we need it**:
Recently https://github.com/kubernetes/kubernetes/pull/44868 is merged which allows using datastore cluster for static and dynamic PVs.

Sample is required in the Read me file to help user, how they can specify path in the yaml for datastore within cluster.

**Which issue this PR fixes**
fixes # https://github.com/vmware/kubernetes/issues/129

**Special notes for your reviewer**:
@BaluDontu @tusharnt

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by cblecker** [comment]
@k8s-bot ok to test

**Comment by divyenpatel** [comment]
@k8s-bot kops aws e2e test this

**Comment by divyenpatel** [comment]
@brendandburns can you approve this PR?

File: examples/volumes/vsphere/README.md [hunk 1]
```diff
@@ -94,6 +94,11 @@
           volumePath: "[datastore1] volumes/myDisk"
           fsType: ext4
       ```
+      In the above example datastore1 not placed within any Folder or Datastore Cluster. If datastore is placed within Datastore Cluster or some folder, appropriate path should be provided in the VolumePath as below.
```
Comment (brendandburns) [inline examples/volumes/vsphere/README.md]: I don't quite follow this sentence. should it be: "datastore1 _is_ not placed.. "?

"_the_ appropriate path"?

File: examples/volumes/vsphere/README.md [hunk 2]
```diff
@@ -233,7 +238,13 @@
       parameters:
           diskformat: zeroedthick
           datastore: VSANDatastore
-      ```
+      ```     
+      If datastore is member of DataStore Cluster or within some folder, appropriate path needs to be provided in the datastore parameter as below.
```
Comment (brendandburns) [inline examples/volumes/vsphere/README.md]: "the appropriate path"?

**Comment by brendandburns** [comment]
@divyenpatel a few comments to clean up the text for clarity.

Also, a quick note on process (we don't document this well, so apologies)

The flow is an lgtm first, and then an approve.

Once the text is cleaned up a little, I'm happy to both lgtm and approve.

Thanks!

File: examples/volumes/vsphere/README.md (hunk 1 above)
Comment (divyenpatel) [inline examples/volumes/vsphere/README.md]: clarified note with mentioning root folder - the datacenter folder

> 
In the above example datastore1 is located in the root folder. If datastore is member of Datastore Cluster or located in sub folder, the folder path needs to be provided in the VolumePath as below

File: examples/volumes/vsphere/README.md (hunk 2 above)
Comment (divyenpatel) [inline examples/volumes/vsphere/README.md]: replaced with "the folder path"

**Pull request merged**